.eslintrc.json

```diff
@@ -1,8 +1,112 @@
 {
-  "extends": ["next/core-web-vitals", "prettier"],
-  "plugins": ["prettier"],
-  "ignorePatterns": ["**/dist/*.js", "/docs/**", "/out/**"],
+  "plugins": ["prettier", "@typescript-eslint", "import"],
+  "parserOptions": {
+    "project": "./tsconfig.json"
+  },
+  "parser": "@typescript-eslint/parser",
```

Comment:
Parsers are required, because `@typescript-eslint/strict-type-checked` uses AST analysis. In general, it makes linting process take a bit longer, but it's not an issue unless codebase is extremely big.